etc/ROS2TEST_QOS_PROFILES.xml

```diff
@@ -20,27 +20,28 @@ xsi:noNamespaceSchemaLocation="http://community.rti.com/schema/5.3.1/rti_dds_qos
           </rtps_well_known_ports>
           <rtps_reserved_port_mask>MASK_DEFAULT</rtps_reserved_port_mask>
         </wire_protocol>
-        <transport_builtin>
+        <!-- <transport_builtin>
           <mask>UDPv4</mask>
-          <!--mask>SHMEM</mask-->
-        </transport_builtin>
-        
+          <mask>SHMEM</mask>
+        </transport_builtin> -->
```

Comment:
I commented out this, because when it's added I have discovery problems, and both transports are the default ones.

I even see warning messages when loading the profile with that uncommented in a rti tool, so it doesn't seem to be a ros related issue.

Reply:
the purpose of having the transport_builtin tag was to test performance of the different transports - the intent was to restrict ROS2 participants to shared memory only for testing, but attempting to do this in ROS2 caused participant creation to fail. 
What warnings did you see? Selecting/restricting the transports is valid.

Reply:
> Selecting/restricting the transports is valid.

I do agree, I couldn't make it work when restricting the transports to only shared memory.
Now, it's working using both SHMEM|UDPv4.

Does the change in the max message size [here](https://github.com/jacobperron/pc_pipe/pull/3#discussion_r407733500) make sense?
I want to check if this is going in the right direction or not.

I will continue investigating why it fails with only SHMEM enabled.

Reply:
I've been testing what's the problem when the available transports is restricted to only `SHMEM`.
What I see is that everything is set up successfully (i.e.: `Participant` and `Endpoint` creation don't fail), and spawn entity is failing to find an available server.

I've debugged the failure, and it seems to me that actually DDS is failing to discover the spawn entity request/reply topics (i.e.: I didn't find any ros related errors, just DDS discovery not returning the information that I expect).
I've also checked that it's not a timing failure (i.e.: I modified spawn entity to try for more time).

I even have sometimes problems to discover those topics from the rti admin console.
More specifically, some instances of the console do discover all the topics well and some others not --with the same example running--.

It seems to me that this is a super fishy discovery problem, but I haven't figured it out.

If I manually launch gazebo with ros plugins, and then manually call spawn entity, it does work well. So, it seems to be affected by the number of entities created at the same time.

I've also tried to restrict discovery to udpv4 and data to shmem (doing something similar to https://community.rti.com/kb/how-do-i-separate-transport-used-discovery-data-transport-used-user-data), without much success either.

Reply:
This sound like something:

```
Note: Connext DDS does not currently support configurations with Docker + SHMEM only. In order for Connext to work properly, you must use Docker + (UDPv4 | SHMEM). This article applies to Connext DDS versions 5.2.0 through 5.3.1
```

from https://community.rti.com/kb/communicate-two-docker-containers-using-rti-connext-dds-and-shared-memory.

I've been using docker all the time.